RevenueCatUI/Views/PurchaseButton.swift

```diff
@@ -116,6 +106,46 @@ struct PurchaseButton: View {
 
 }
 
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, *)
+private struct PurchaseButtonLabel: View {
+
+    let package: TemplateViewConfiguration.Package
+    let colors: PaywallData.Configuration.Colors
+    let introEligibility: IntroEligibilityStatus?
+    let mode: PaywallViewMode
+
+    @Environment(\.userInterfaceIdiom)
+    private var userInterfaceIdiom
+
+    @Environment(\.isEnabled)
+    private var isEnabled
+
+    var body: some View {
+        IntroEligibilityStateView(
+            textWithNoIntroOffer: self.package.localization.callToAction,
+            textWithIntroOffer: self.package.localization.callToActionWithIntroOffer,
+            introEligibility: self.introEligibility,
+            foregroundColor: self.colors.callToActionForegroundColor
+        )
+            .frame(
+                maxWidth: self.mode.fullWidthButton
+                   ? .infinity
+                    : nil
+            )
+            .padding()
+            .padding(.vertical, self.userInterfaceIdiom == .pad ? 10 : 0)
+            .hidden(if: !self.isEnabled)
+            .overlay {
+                if !self.isEnabled {
+                    ProgressView()
```

Comment:
Super minor, FFTI: but given that it goes into a loading state when it's not enabled, maybe the flag should be something like `isLoading` or `hasOperationInProgress` or something to reflect that

Reply:
But it's coming from the `.isEnabled` environment variable, that's why I named it as such 🤔